### PR TITLE
Prop cleanup

### PR DESCRIPTION
**Pull request by BSd3v**
- prop clean-up overhaul
- updating testing that broke due to props no longer being on the grid, but in `dashGridOptions`
- added testing for selection and sizing buttons
- removed `cellStyle` from the grid level, allowing more flexibility in customization, and alignment with AG grid
- allowed for functions, styleConditions and regular dictionaries to be passed to the `cellStyle` on all levels
- added `rowId` to `cellClicked` data
- added tests for `cellClicked` data

### CODE REVIEW

**Comment by BSd3v** [comment]
I am going to change `nodeId` to `rowId` as well for `cellValueChanged`, that way it lines up.

@AnnMarieW, @alexcjohnson

File: src/lib/components/AgGrid.react.js [hunk 1]
```diff
@@ -153,28 +149,20 @@ DashAgGrid.propTypes = {
...
      * If true, the internal method selectAll() will be called
+     * If object, then the object will be passed to selectAll()
      */
-    selectAll: PropTypes.bool,
-
-    /**
-     * If true, the internal method selectAllFiltered() will be called
-     */
-    selectAllFiltered: PropTypes.bool,
+    selectAll: PropTypes.oneOfType([PropTypes.bool, PropTypes.object]),
```
Comment (alexcjohnson) [inline src/lib/components/AgGrid.react.js:154]:
```suggestion
    selectAll: PropTypes.oneOfType([
        PropTypes.bool,
        PropTypes.exact({
            filtered: PropTypes.bool
        })
    ]),
```

File: src/lib/components/AgGrid.react.js [hunk 2]
```diff
@@ -153,28 +149,20 @@ DashAgGrid.propTypes = {
 
     /**
      * If true, the internal method selectAll() will be called
+     * If object, then the object will be passed to selectAll()
```
Comment (alexcjohnson) [inline src/lib/components/AgGrid.react.js:152]:
```suggestion
     * Set to true to cause all rows to be selected,
     * Or pass an object of options for which rows to select.
     * Currently supports `filtered`, set to true to only select filtered rows.
```

File: src/lib/components/AgGrid.react.js [hunk 3]
```diff
@@ -153,28 +149,20 @@ DashAgGrid.propTypes = {
...
+     * If true, the internal method autoSizeAllColumns(false) will be called to not skip headers
+     * If object, then the object will be passed to autoSizeAllColumns
      */
-    autoSizeAllColumns: PropTypes.bool,
-
-    /**
-     * If true, the internal method autoSizeAllColumns(true) will be called
-     */
-    autoSizeAllColumnsSkipHeaders: PropTypes.bool,
+    autoSizeAllColumns: PropTypes.oneOfType([PropTypes.bool, PropTypes.object]),
```
Comment (alexcjohnson) [inline src/lib/components/AgGrid.react.js:165]:
```suggestion
    autoSizeAllColumns: PropTypes.oneOfType([
        PropTypes.bool,
        PropTypes.exact({
            skipHeaders: PropTypes.bool
        })
    ]),
```

File: src/lib/components/AgGrid.react.js [hunk 4]
```diff
@@ -153,28 +149,20 @@ DashAgGrid.propTypes = {
...
 
     /**
      * If true, the internal method deselectAll() will be called
      */
     deselectAll: PropTypes.bool,
 
     /**
-     * If true, the internal method autoSizeAllColumns(false) will be called
+     * If true, the internal method autoSizeAllColumns(false) will be called to not skip headers
+     * If object, then the object will be passed to autoSizeAllColumns
```
Comment (alexcjohnson) [inline src/lib/components/AgGrid.react.js:163]:
```suggestion
     * Set to true to autosize all columns, considering both headers and data
     * Or pass an object of options for how to autosize.
     * Currently supports `skipHeaders`, set this true to only consider data when autosizing.
```

File: src/lib/components/AgGrid.react.js [hunk 5]
```diff
@@ -458,1559 +439,153 @@ DashAgGrid.propTypes = {
...
-     * Default Value: false
-     */
-    rowDragManaged: PropTypes.bool,
+    defaultColDef: PropTypes.object,
 
     /**
-     * Set to true to suppress Row Dragging.
-     * Default Value: false
+     * Sets the Row Model type.
+     * Default Value: ['clientSide', 'infinite', 'viewport', 'serverSide']
```
Comment (alexcjohnson) [inline src/lib/components/AgGrid.react.js:451]:
```suggestion
     * Default Value: 'clientSide'
```
I know this isn't new in this PR, just a git artifact... but it drew my attention.

**Review by alexcjohnson — APPROVED** [review]
Excellent, much cleaner! 💃

**Pull request merged**